networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/AttestationGossipManager.java

```diff
@@ -63,25 +69,41 @@ public void onNewAttestation(final ValidateableAttestation validateableAttestati
               attestationPublishSuccessCounter.inc();
             },
             error -> {
-              if (Throwables.getRootCause(error) instanceof MessageAlreadySeenException) {
-                LOG.debug(
-                    "Failed to publish attestation for slot {} because the message has already been seen",
-                    attestation.getData().getSlot());
-              } else if (Throwables.getRootCause(error)
-                  instanceof NoPeersForOutboundMessageException) {
-                LOG.warn(
-                    "Failed to publish attestation for slot {} because no peers were available on the required gossip topic",
-                    attestation.getData().getSlot());
-              } else {
-                LOG.error(
-                    "Failed to publish attestation for slot {}",
-                    attestation.getData().getSlot(),
-                    error);
-              }
+              logWithSuppression(error, attestation);
               attestationPublishFailureCounter.inc();
             });
   }
 
+  synchronized void logWithSuppression(final Throwable error, final Attestation attestation) {
+    final AttestationData attestationData = attestation.getData();
+    final Throwable rootCause = Throwables.getRootCause(error);
+    if (attestationData.getSlot().equals(lastErroredSlot)
+        && attestationData.getIndex().equals(lastErroredCommitteeIndex)
+        && rootCause.equals(lastRootCause)) {
```

Comment:
Slightly suspicious of using `.equals` on an Exception. I wonder if it would be better to just check if the root cause class is the same since we're only selecting based on the `instanceof` conditions below anyway.

Reply:
Oh and probably worth logging at debug level instead of entirely suppressing.